pkg/component/worker/containerd.go

```diff
@@ -59,88 +60,136 @@ imports = [
+		c.importsPath = importsPathWindows
+	} else {
+		c.binaries = []string{"containerd", "containerd-shim", "containerd-shim-runc-v1", "containerd-shim-runc-v2", "runc"}
+		c.confPath = confPathPosix
+		c.importsPath = importsPathPosix
+	}
+	return c
+}
+
 var _ manager.Component = (*ContainerD)(nil)
 
 // Init extracts the needed binaries
 func (c *ContainerD) Init(ctx context.Context) error {
 	g, _ := errgroup.WithContext(ctx)
-	for _, bin := range []string{"containerd", "containerd-shim", "containerd-shim-runc-v1", "containerd-shim-runc-v2", "runc"} {
+	for _, bin := range c.binaries {
 		b := bin
 		g.Go(func() error {
 			return assets.Stage(c.K0sVars.BinDir, b, constant.BinDirMode)
 		})
 	}
-
+	if err := c.windowsInit(); err != nil {
+		return fmt.Errorf("windows init failed: %v", err)
+	}
 	return g.Wait()
 }
 
+func (c *ContainerD) windowsInit() error {
+	if runtime.GOOS != "windows" {
+		return nil
+	}
+	// On windows we need always run containerd.exe as a service
+	command := fmt.Sprintf("if (-not (Get-Service -Name containerd -ErrorAction SilentlyContinue)) { %s\\containerd.exe --register-service}", c.K0sVars.BinDir)
+	return winExecute(command)
+}
+
 // Run runs containerD
 func (c *ContainerD) Start(ctx context.Context) error {
 	logrus.Info("Starting containerD")
 
 	if err := c.setupConfig(); err != nil {
 		return fmt.Errorf("failed to setup containerd config: %w", err)
 	}
+	if runtime.GOOS == "windows" {
+		if err := c.windowsStart(ctx); err != nil {
+			return fmt.Errorf("failed to start windows server: %v", err)
+		}
+	} else {
+		c.supervisor = supervisor.Supervisor{
+			Name:    "containerd",
+			BinPath: assets.BinPath("containerd", c.K0sVars.BinDir),
+			RunDir:  c.K0sVars.RunDir,
+			DataDir: c.K0sVars.DataDir,
+			Args: []string{
+				fmt.Sprintf("--root=%s", filepath.Join(c.K0sVars.DataDir, "containerd")),
+				fmt.Sprintf("--state=%s", filepath.Join(c.K0sVars.RunDir, "containerd")),
+				fmt.Sprintf("--address=%s", filepath.Join(c.K0sVars.RunDir, "containerd.sock")),
+				fmt.Sprintf("--log-level=%s", c.LogLevel),
+				fmt.Sprintf("--config=%s", c.confPath),
+			},
+		}
 
-	c.supervisor = supervisor.Supervisor{
-		Name:    "containerd",
-		BinPath: assets.BinPath("containerd", c.K0sVars.BinDir),
-		RunDir:  c.K0sVars.RunDir,
-		DataDir: c.K0sVars.DataDir,
-		Args: []string{
-			fmt.Sprintf("--root=%s", filepath.Join(c.K0sVars.DataDir, "containerd")),
-			fmt.Sprintf("--state=%s", filepath.Join(c.K0sVars.RunDir, "containerd")),
-			fmt.Sprintf("--address=%s", filepath.Join(c.K0sVars.RunDir, "containerd.sock")),
-			fmt.Sprintf("--log-level=%s", c.LogLevel),
-			fmt.Sprintf("--config=%s", confPath),
-		},
-	}
-
-	if err := c.supervisor.Supervise(); err != nil {
-		return err
+		if err := c.supervisor.Supervise(); err != nil {
+			return err
+		}
 	}
 
 	go c.watchDropinConfigs(ctx)
 
 	return nil
 }
 
+func (c *ContainerD) windowsStart(_ context.Context) error {
+	return winExecute("Start-Service containerd")
```

Comment:
Maybe wrap the error here instead of at the call site at line 134? (Same for the `windowsStop` method.)

```suggestion
	if err := winExecute("Start-Service containerd"); err != nil {
		fmt.Errorf("failed to start Windows Service %q: %w", "containerd", err)
	}
	return nil
```